src/Json.re

```diff
@@ -236,16 +245,50 @@ let parseString = (text, pos) => {
   (Scanf.unescaped(String.sub(text, pos, i^ - pos)), i^ + 1)
 };
 
-let parseNumber = (text, pos) => {
+let rec toCharList = (text, maxDepth) => {
+  if (String.length(text) < 2 || maxDepth < 2) {
+    [text.[0]]
+  } else {
+    [text.[0], ...toCharList(stringTail(text), maxDepth - 1)]
+  }
+};
+
+let continousDigits = (text, pos, len) => {
   let i = ref(pos);
-  let len = String.length(text);
-  while (i^ < len && Char.code('0') <= Char.code(text.[i^]) && Char.code(text.[i^]) <= Char.code('9')) {
+  let dec = ref(false);
+  let isNumber = n =>
+    Char.code('0') <= Char.code(text.[n]) && Char.code(text.[n]) <= Char.code('9');
+  let isDec = n =>
+    dec^ == false && Char.code('.') == Char.code(text.[n]);
```

Comment:
Why the char.code here? can't you do `'.' == text.[n]`?

Reply:
True. Remnants from earlier. Fixed.